src/v/raft/consensus.cc

```diff
@@ -158,7 +161,11 @@ ss::future<> consensus::stop() {
     return _event_manager.stop()
       .then([this] { return _append_requests_buffer.stop(); })
       .then([this] { return _batcher.stop(); })
-      .then([this] { return _bg.close(); })
+      .then([this] {
+          auto f = _bg.close();
```

Comment:
I'm not sure I follow why we wait for bg after the follower queues have shutdown. If bg closes it seems like you could assert in follower_queues::stop that all the semaphores are free.